### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by seandelacerda**

### CODE REVIEW

File: src/main/java/com/cst438/controller/StudentController.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
+package com.cst438.controller;
```
Comment (seandelacerda) [inline src/main/java/com/cst438/controller/StudentController.java:1]: substantive changes are to this file

File: src/main/java/com/cst438/domain/CourseRepository.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.query.Param;
+
+public interface CourseRepository extends CrudRepository <Course, Integer> {
+	
+	@Query("select c from Course c where c.course_id=:course_id")
+	public Course findByCourse_id(@Param("course_id")int course_id);
```
Comment (seandelacerda) [inline src/main/java/com/cst438/domain/CourseRepository.java:11]: had to add @Param here to get code running

File: src/main/java/com/cst438/domain/EnrollmentRepository.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import java.util.List;
+
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.query.Param;
+
+public interface EnrollmentRepository extends CrudRepository <Enrollment, Integer> {
+	
+	@Query("select e from Enrollment e where e.enrollment_id=:id")
+	Enrollment findById(@Param("id")int id);
```
Comment (seandelacerda) [inline src/main/java/com/cst438/domain/EnrollmentRepository.java:13]: added @Param to these as well

File: src/main/java/com/cst438/controller/StudentController.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+import com.cst438.domain.StudentRepository;
+import com.cst438.service.GradebookService;
+
+@RestController
+@CrossOrigin(origins = "http://localhost:3000")
+public class StudentController {
+   @Autowired
+   StudentRepository studentRepository;
+   
+   // create a new student record
```
Comment (ejrollo) [inline src/main/java/com/cst438/controller/StudentController.java:34]: good implementation of request parameters

File: src/main/java/com/cst438/controller/StudentController.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+      Student student = studentRepository.findByEmail(email);
+      
+      if (student != null) {
+         // student exists, add hold
+         // statuses: 1 = hold, 0 = clear
+         student.setStatusCode(1);
+         student.setStatus("Hold"); 
+         
+         studentRepository.save(student);
+
```
Comment (ejrollo) [inline src/main/java/com/cst438/controller/StudentController.java:71]: good use of throwing exceptions

File: src/main/java/com/cst438/controller/StudentController.java [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+      
+      if (student != null) {
+         // student exists, remove hold
+         // statuses: 1 = hold, 0 = clear
+         student.setStatusCode(0);
+         student.setStatus(null); 
+         
+         studentRepository.save(student);
+         
+         return student;
```
Comment (ejrollo) [inline src/main/java/com/cst438/controller/StudentController.java:95]: good use of exceptions.

**Review by ejrollo — APPROVED** [review]
Overall good method implementations and good use of throwing exceptions if data not found.

**Pull request merged**